packages/material-ui/src/InputBase/InputBase.test.js

```diff
@@ -182,6 +182,37 @@ describe('<InputBase />', () => {
       expect(typeof injectedProps.onBlur).to.equal('function');
       expect(typeof injectedProps.onFocus).to.equal('function');
     });
+
+    it('throws on change if `inputRef` isnt forwarded', () => {
+      const BadInputComponent = React.forwardRef(({ action, inputRef, ...props }, ref) => {
+        React.useImperativeHandle(action, () => ({
+          trigger: () => {
+            props.onChange({}, {});
+          },
+        }));
+
+        // `inputRef` belongs into `ref`
+        return <input {...props} ref={ref} />;
+      });
+
+      BadInputComponent.propTypes = {
+        action: PropTypes.object.isRequired,
+        inputRef: PropTypes.func.isRequired,
+        onChange: PropTypes.func.isRequired,
+      }
+
+      const handleRef = React.createRef();
+      render(
+        <InputBase inputProps={{ action: handleRef }} inputComponent={BadInputComponent} />,
+      );
+      let errorMessage = null;
+      try {
+        handleRef.current.trigger();
```

Comment:
fireEvent.change needs a non-null value to trigger the handler.

Reply:
Right. That was the part I was missing about change events. Got a simpler idea though which is closer to how react-stripe-elements works.

Reply:
The issue was less about how change events are created but rather about error handling in react. I'm unable to catch errors from event handlers in our browser tests.